database/sqlite/sqlite_context.c

```diff
@@ -0,0 +1,597 @@
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to prepare statement to store context");
+        return 1;
+    }
+
+    rc = sqlite3_bind_blob(res, 1, *host_uuid, sizeof(*host_uuid), SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind host_uuid to store context details");
+        goto failed;
+    }
+
+    rc = sqlite3_bind_text(res, 2, context_data->id, -1, SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind context to store details");
+        goto failed;
+    }
+
+    rc = sqlite3_bind_int64(res, 3, (time_t) context_data->version);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind first_time_t to store context details");
+        goto failed;
+    }
+
+
+    rc = bind_text_null(res, 4, context_data->title);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind context to store details");
+        goto failed;
+    }
+
+    rc = bind_text_null(res, 5, context_data->chart_type);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind context to store details");
+        goto failed;
+    }
+
+    rc = bind_text_null(res, 6, context_data->units);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind context to store details");
+        goto failed;
+    }
+
+    rc = sqlite3_bind_int64(res, 7, (time_t) context_data->priority);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind first_time_t to store context details");
+        goto failed;
+    }
+
+    rc = sqlite3_bind_int64(res, 8, (time_t) context_data->first_time_t);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind first_time_t to store context details");
+        goto failed;
+    }
+
+    rc = sqlite3_bind_int64(res, 9, (time_t) context_data->last_time_t);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind last_time_t to store context details");
+        goto failed;
+    }
+
+    rc = sqlite3_bind_int(res, 10, (time_t) context_data->deleted);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind last_time_t to store context details");
+        goto failed;
+    }
+
+    rc = sqlite3_bind_text(res, 11, context_data->family, -1, SQLITE_STATIC);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("Failed to bind context to store details");
+        goto failed;
+    }
+
+    rc_stored = execute_insert(res);
+
+    if (rc_stored != SQLITE_DONE)
+        error_report("Failed store context details for context %s, rc = %d", context_data->id, rc_stored);
+
+failed:
+    rc = sqlite3_finalize(res);
+    if (rc != SQLITE_OK)
+        error_report("Failed to finalize statement that stores context details, rc = %d", rc);
+
+    return (rc_stored != SQLITE_DONE);
+}
+
+// Delete a context
+
+#define CTX_DELETE_CONTEXT "DELETE FROM context WHERE host_id = @host_id AND id = @context;"
```

Comment:
When we execute this kind of SQL command, we are also losing some data, I suggest either to have a table to store hosts that are removed or to store an additional information inside log when we remove hosts on success.

Reply:
We can have this only when `NETDATA_INTERNAL_CHECKS` is set.

Reply:
Will report in the log in this case